apps/projects/app/components/Shared/FilterBar/FilterBar.js

```diff
@@ -299,7 +302,7 @@ class FilterBar extends React.Component {
 
           <FilterDropDown
             caption="Status"
-            enabled={true}
+            enabled={Object.keys(filtersData.milestones).length > 0}
```

Comment:
Could `filtersData` ever be non-existent? Code like this _should_ be fine, but if `filtersData` ever disappears this is going to throw an error like: "cannot read property milestones of undefined" (this caused some errors in other parts of the app due to the script.js file overwriting state  causing certain variables to disappear)

Maybe consider the safe bet and do:

`filtersData && Object.keys(filtersData.milestones).length > 0`

But that could be overkill

Reply:
For example, take a look at this line: https://github.com/AutarkLabs/planning-suite/blob/107a67f1c23eb37eb45268e1a0ceaf6c1442223d/apps/projects/app/components/Shared/FilterBar/FilterBar.js#L85

this appears to be mutating state, which can have some weird side effects (unless ES6 makes a clone of the state variable and not a reference to it).

Just something to be aware of

Reply:
Ok, second one improved, first one... in theory there is no possibility filtersData should not be there. everything relies on it. So for now I'm going to leave it as it is.